test/evaluate/ISO-Fortran-binding.cc

```diff
@@ -210,14 +210,459 @@ int main() {
       rank3d, extents);
   MATCH(false,
       dv_3darray->dim[2].extent == 2 + 66);  // extents was read
+}
+
+static void check_CFI_address(
+    const CFI_cdesc_t *dv, const CFI_index_t subscripts[]) {
+  // 18.5.5.2
+  void *addr = CFI_address(dv, subscripts);
+  const Descriptor *desc = reinterpret_cast<const Descriptor *>(dv);
+  void *addrCheck = desc->Element<void>(subscripts);
+  MATCH(true, addr == addrCheck);
+}
+
+// Helper function to set lower bound of descriptor
+static void EstablishLowerBounds(CFI_cdesc_t *dv, CFI_index_t *sub) {
+  for (int i{0}; i < dv->rank; ++i) {
+    dv->dim[i].lower_bound = sub[i];
+  }
+}
+
+// Helper to get size without making internal compiler functions accessible
+static std::size_t ByteSize(CFI_type_t ty, std::size_t size) {
+  CFI_CDESC_T(0) storage;
+  CFI_cdesc_t *dv = reinterpret_cast<CFI_cdesc_t *>(&storage);
```

Comment:
reinterpret_cast<T> should be used as a last resort.

Reply:
Is it even necessary here?  If so, could the CFI_CDESC_T macro be tweaked to avoid it?

Also: braces, please.

Reply:
Currently, it is needed.
I have changed CFI_CDESC_T macro in C++ to avoid this. It now inherits from CFI_cdesc_t so that there is an implicit conversion to the base class and no cast is needed anymore.